src/library/librarycontrol.cpp

```diff
@@ -246,6 +257,10 @@ void LibraryControl::sidebarWidgetDeleted() {
     m_pSidebarWidget = nullptr;
 }
 
+void LibraryControl::searchBoxWidgetDeleted() {
+    m_pSearchBoxWidget = nullptr;
```

Comment:
I think this can be done behind the scenes with QPointer
https://doc.qt.io/qt-5/qpointer.html

Reply:
right now I don't need this anymore because I'd like to connect a signal to WSearchLineEdit::clearSearch.
A lot of trial and error again, and I learn more about C++ and Qt :)